rs/index_server/src/collection_manager.rs

```diff
@@ -273,4 +297,31 @@ impl CollectionManager {
         let hash = hasher.finish();
         hash as u32 % num_workers
     }
+
+    pub async fn get_consumer_by_topic(&self, topic_name: &str) -> MutexGuard<'_, LogConsumer> {
```

Comment:
I'm not a fan of exposing the internal lock with `pub`. Easy to misuse. Remove the `pub`